src/lib/cli/progress.js

```diff
@@ -0,0 +1,96 @@
+// @flow
+
+/**
+ * External dependencies
+ */
+import chalk from 'chalk';
+import { stdout as progressLog } from 'single-line-log';
+
+// Progress spinner
+const sprite = {
+	i: 0,
+	sprite: [ '⠋', '⠙', '⠹', '⠸', '⠼', '⠴', '⠦', '⠧', '⠇', '⠏' ],
+	next() {
+		this.i++;
+
+		if ( this.i >= this.sprite.length ) {
+			this.i = 0;
+		}
+
+		return {
+			value: this.sprite[ this.i ],
+			done: false,
+		};
+	},
+};
+
+// Icons for different action states/status
+const marks = {
+	pending: '○',
+	running: chalk.blueBright( sprite.next().value ),
+	success: chalk.green( '✓' ),
+	failed: chalk.red( '✕' ),
+	skipped: chalk.green( '✕' ),
+};
+
+// Various action steps for SQL imports
+const steps = {
+	replace: 'Performing Search and Replace',
+	validate: 'Validating SQL',
+	upload: 'Uploading file to S3',
+	startImport: 'Starting import',
+	import: 'Importing...',
+	// Add more as needed
+};
```

Comment:
These strings are specific to imports, but the file otherwise seems generalized.

Reply:
Yup, I'll generalize these and consolidate the sprite stuff from master + this branch + progress logic from data sync) in another PR.